### ISSUE TITLE
I couldn't  get correct result when using focal loss 

### ISSUE THREAD

**Issue by zhizhengyang** [open-mmlab/mmdetection#784]
when i use focal loss in cascade_rcnn_dconv_c3-c5_r50_fpn_1x.py with coco pretrained model,the training acc only 0.23,and the result mAP only 0.034.but its correct when using crossentropy loss.I try to change weighted_sigmoid_focal_loss,I get trainging acc 88,but i still get wrong result,the mAP only 3.4,i want  to know how to add right focal loss in cascade rcnn dcn,thanks a lot.
```
def weighted_sigmoid_focal_loss(pred,
                                target,
                                weight,
                                gamma=2.0,
                                alpha=0.25,
                                avg_factor=None,
                                reduce = False,
                                num_classes=21):
    if avg_factor is None:
        avg_factor = torch.sum(weight > 0).float().item() / num_classes + 1e-6

    if pred.dim() != target.dim():
        target, weight = _expand_binary_labels(target, weight, num_classes)

    focal_loss = sigmoid_focal_loss(pred, target, gamma, alpha, 'none')
    # print(focal_loss.shape, weight.shape)

    # res = torch.sum(focal_loss * weight) / avg_factor
    # print(res.shape)

    return torch.sum(focal_loss * weight.float())[None] / avg_factor`
```

**Comment by zhizhengyang** [comment]
here is correct format code.I use this still get wrong result.
```
def weighted_sigmoid_focal_loss(pred,
                                target,
                                weight,
                                gamma=2.0,
                                alpha=0.25,
                                avg_factor=None,
                                reduce = False,
                                num_classes=20):
    if pred.dim() != target.dim():
        target, weight = _expand_binary_labels(target, weight, target.size(-1))
    if avg_factor is None:
        avg_factor = torch.sum(weight > 0).float().item() / num_classes + 1e-6
    return torch.sum(sigmoid_focal_loss(pred, target, gamma, alpha, 'none') * weight.view(-1, 1))[None] / avg_factor
```

**Comment by ztyxd** [comment]
I also have problems with this question, can u solve this problem or we can talk about with this question

**Comment by zhizhengyang** [comment]
> I also have problems with this question, can u solve this problem or we can talk about with this question

i try a lot of times but still couldn't sovle this problem

**Comment by ztyxd** [comment]
@zhizhengyang  Thanks a lot. My QQ number is 810289611 may be we can talk about this question
